### PR TITLE
Make gsutil work when gcloud is authenticated with a ServiceAccount

### PR DESCRIPTION
**Pull request by huyhg**

### CODE REVIEW

**Comment by vcanaa** [comment]
Can you add some context in the description. This a GCP service account, right? This change allows using a service account key to authenticate with gcloud?

File: marketplace/dev/bin/setup_config.sh [hunk 1]
```diff
@@ -44,3 +44,11 @@ if [[ -e "/mount/config/.config/gcloud" ]]; then
   rm -rf "$HOME/.config/gcloud"
   cp -r "/mount/config/.config/gcloud" "$HOME/.config"
 fi
+
+if [[ "${GCLOUD_ORIGINAL_PATH}" != "" ]]; then
+  # Replace gcloud path prefixes with the one mounted in this container.
+  # .boto files are the configuration files for gsutil.
+  find "$HOME/.config" -name ".boto" \
+    | xargs -n 1 sed -i "s|gs_service_key_file = ${GCLOUD_ORIGINAL_PATH}/|gs_service_key_file = $HOME/.config/gcloud/|"
```
Comment (wgrzelak) [inline marketplace/dev/bin/setup_config.sh]:
```suggestion
    | xargs -n 1 sed -i "s|^gs_service_key_file = ${GCLOUD_ORIGINAL_PATH}/|gs_service_key_file = $HOME/.config/gcloud/|"
```

**Comment by huyhg** [comment]
@vcanaa Context in b/135451991.

File: scripts/dev [hunk 2]
```diff
@@ -18,16 +18,23 @@ elif [[ -d "${KUBE_CONFIG}" ]]; then
...
 fi
+
 gcloud_mount=""
+# gcloud_original_path is used for determining the prefix to filenames referenced
+# by certain config files. Such prefix must be replaced by the corresponding
+# mount path in the container.
+gcloud_original_path=""
 if [[ -e "${GCLOUD_CONFIG}" ]]; then
   gcloud_mount=(--mount "type=bind,source=${GCLOUD_CONFIG},target=/mount/config/.config/gcloud,readonly")
+  gcloud_original_path=(--env "GCLOUD_ORIGINAL_PATH=$(readlink -f ${GCLOUD_CONFIG})")
```
Comment (wgrzelak) [inline scripts/dev]: @huyhg `readlink -f` doesn't work on iOS. Could you take a look?

Comment (huyhg) [reply scripts/dev]: Fixed by https://github.com/GoogleCloudPlatform/marketplace-k8s-app-tools/pull/380

**Pull request merged**